File: rnmodules/react-native-kb/android/src/oldarch/KbSpec.java
package com.reactnativekb;

import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReadableMap;

abstract class KbSpec extends ReactContextBaseJavaModule {
  KbSpec(ReactApplicationContext context) {
    super(context);
  }

    public abstract void getDefaultCountryCode(Promise promise);
    public abstract void logSend(String status, String feedback, boolean sendLogs, boolean sendMaxBytes, String traceDir, String cpuProfileDir, Promise promise);
    public abstract void androidOpenSettings();
    public abstract void androidSetSecureFlagSetting(boolean setSecure, Promise promise);
    public abstract void androidGetSecureFlagSetting(Promise promise);
    public abstract void androidShare(String uriPath, String mimeType, Promise promise);
    public abstract void androidShareText(String text, String mimeType, Promise promise);
    public abstract void androidCheckPushPermissions(Promise promise);
    public abstract void androidRequestPushPermissions(Promise promise);
    public abstract void androidGetRegistrationToken(Promise promise);
    public abstract void androidUnlink(String path, Promise promise);
    public abstract void androidAddCompleteDownload(ReadableMap config, Promise promise);
    public abstract void androidAppColorSchemeChanged(String prefString);
    public abstract void androidSetApplicationIconBadgeNumber(double badge);
    public abstract void androidGetInitialBundleFromNotification(Promise promise);
    public abstract void androidGetInitialShareFileUrl(Promise promise);
    public abstract void androidGetInitialShareText(Promise promise);
    public abstract void engineReset();
    public abstract void engineStart();
    public abstract void rpcOnGo(byte[] arr);
}

File: rnmodules/react-native-kb/android/src/oldarch/KbSpec.kt
package com.reactnativekb

import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReadableMap
import java.util.Map

abstract class KbSpec(context: ReactApplicationContext?) : ReactContextBaseJavaModule(context) {

    abstract fun getTypedExportedConstants(): MutableMap<String, Any>
    abstract fun install(): Boolean
    abstract fun getDefaultCountryCode(promise: Promise)
    abstract fun logSend(status: String, feedback: String, sendLogs: Boolean, sendMaxBytes: Boolean, traceDir: String, cpuProfileDir: String, promise: Promise)
    abstract fun androidOpenSettings()
    abstract fun androidSetSecureFlagSetting(setSecure: Boolean, promise: Promise)
    abstract fun androidGetSecureFlagSetting(promise: Promise)
    abstract fun androidShare(uriPath: String, mimeType: String, promise: Promise)
    abstract fun androidShareText(text: String, mimeType: String, promise: Promise)
    abstract fun androidCheckPushPermissions(promise: Promise)
    abstract fun androidRequestPushPermissions(promise: Promise)
    abstract fun androidGetRegistrationToken(promise: Promise)
    abstract fun androidUnlink(path: String, promise: Promise)
    abstract fun androidAddCompleteDownload(config: ReadableMap, promise: Promise)
    abstract fun androidAppColorSchemeChanged(prefString: String)
    abstract fun androidSetApplicationIconBadgeNumber(badge: Double)
    abstract fun androidGetInitialShareFileUrls(promise: Promise)
    abstract fun engineReset()
    abstract fun engineStart()
    abstract fun shareListenersRegistered()
    abstract fun addListener(eventType: String)
    abstract fun removeListeners(count: Double)
    abstract fun iosGetHasShownPushPrompt(promise: Promise)

}
